client/document/index.jsx

```diff
@@ -100,16 +103,29 @@ class Document extends Component {
 
 		const isRTL = isLocaleRtl( lang );
 
+		const redirectToParams =
+			typeof query?.redirect_to === 'string' && query.redirect_to.split( '?' )[ 1 ];
+
+		// Get the client ID from the redirect URL to cover the case of a login URL without the "client_id" parameter.
+		// e.g. /log-in/link/use
+		const clientId = new URLSearchParams( redirectToParams ).get( 'client_id' );
```

Comment:
Previously, we utilized the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL) to obtain the `client_id`. This API requires the input to be in HTTP URL format; otherwise, it will throw a JavaScript fatal error. However, the `query.redirect_to` of the VideoPress's signup page does not conform to this format.

Now, we have adopted the use of [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams), which boasts high error tolerance, thereby ensuring that the page will not break during Server Side Rendering (SSR).

> Note: We can validate the new logic by visiting [VideoPress's signup page](http://calypso.localhost:3000/start/videopress-account/user/en?variationName=videopress&flow=videopress&pageTitle=Video%20Portfolio&redirect_to=/setup/videopress/options), it should still work.